### ISSUE TITLE
Conflicting namespaces on OSX with hxcpp

### ISSUE THREAD

**Issue by Aidan63** [peteshand/signals#3]
Following on from my PR (#2) the following code sample will not compile on OSX High Sierra with hxcpp git and haxe 4.0.0-rc.2

```haxe
import signal.Signal;

class Main
{
    static function main()
    {
        var signal = new Signal();
        signal.add(() -> trace('hello world!'));
        signal.dispatch();
    }
}
```
You get the following error log.
```
Error: In file included from ./src/signal/Signal.cpp:5:
include/signal/BaseSignal.h:10:19: error: redefinition of 'signal' as different kind of symbol
HX_DECLARE_CLASS1(signal,BaseSignal)
                  ^
/usr/include/sys/signal.h:390:8: note: previous definition is here
void    (*signal(int, void (*)(int)))(int);
          ^
In file included from ./src/signal/Signal.cpp:5:
include/signal/BaseSignal.h:12:11: error: redefinition of 'signal' as different kind of symbol
namespace signal{
          ^
/usr/include/sys/signal.h:390:8: note: previous definition is here
void    (*signal(int, void (*)(int)))(int);
          ^
In file included from ./src/signal/Signal.cpp:5:
include/signal/BaseSignal.h:43:29: error: no member named 'BaseSignal' in namespace 'signal'
HXLINE(  44)            ( ( ::signal::BaseSignal)(__this) )->priorityUsed = false;
                            ~~~~~~~~~~^
include/signal/BaseSignal.h:44:29: error: no member named 'BaseSignal' in namespace 'signal'
HXLINE(  43)            ( ( ::signal::BaseSignal)(__this) )->requiresSort = false;
                            ~~~~~~~~~~^
include/signal/BaseSignal.h:45:29: error: no member named 'BaseSignal' in namespace 'signal'
HXLINE(  42)            ( ( ::signal::BaseSignal)(__this) )->toTrigger = ::cpp::VirtualArray_obj::__new(0);
                            ~~~~~~~~~~^
include/signal/BaseSignal.h:46:29: error: no member named 'BaseSignal' in namespace 'signal'
HXLINE(  41)            ( ( ::signal::BaseSignal)(__this) )->callbacks = ::Array_obj< ::Dynamic>::__new(0);
                            ~~~~~~~~~~^
include/signal/BaseSignal.h:47:29: error: no member named 'BaseSignal' in namespace 'signal'
HXLINE(  40)            ( ( ::signal::BaseSignal)(__this) )->fireOnAdd = false;
                            ~~~~~~~~~~^
include/signal/BaseSignal.h:48:29: error: no member named 'BaseSignal' in namespace 'signal'
HXLINE(  48)            ( ( ::signal::BaseSignal)(__this) )->fireOnAdd = fireOnAdd;
                            ~~~~~~~~~~^
In file included from ./src/signal/Signal.cpp:8:
include/signal/Signal.h:13:19: error: redefinition of 'signal' as different kind of symbol
HX_DECLARE_CLASS1(signal,BaseSignal)
                  ^
/usr/include/sys/signal.h:390:8: note: previous definition is here
void    (*signal(int, void (*)(int)))(int);
          ^
In file included from ./src/signal/Signal.cpp:8:
include/signal/Signal.h:14:19: error: redefinition of 'signal' as different kind of symbol
HX_DECLARE_CLASS1(signal,Signal)
                  ^
/usr/include/sys/signal.h:390:8: note: previous definition is here
void    (*signal(int, void (*)(int)))(int);
          ^
In file included from ./src/signal/Signal.cpp:8:
include/signal/Signal.h:16:11: error: redefinition of 'signal' as different kind of symbol
namespace signal{
          ^
/usr/include/sys/signal.h:390:8: note: previous definition is here
void    (*signal(int, void (*)(int)))(int);
          ^
./src/signal/Signal.cpp:14:11: error: redefinition of 'signal' as different kind of symbol
namespace signal{
          ^
/usr/include/sys/signal.h:390:8: note: previous definition is here
void    (*signal(int, void (*)(int)))(int);
          ^
./src/signal/Signal.cpp:16:6: error: use of undeclared identifier 'Signal_obj'
void Signal_obj::__construct( ::Dynamic fireOnAdd){
     ^
./src/signal/Signal.cpp:21:9: error: use of undeclared identifier 'Signal_obj'
Dynamic Signal_obj::__CreateEmpty() { return new Signal_obj; }
        ^
./src/signal/Signal.cpp:23:7: error: use of undeclared identifier 'Signal_obj'
void *Signal_obj::_hx_vtable = 0;
      ^
./src/signal/Signal.cpp:25:9: error: use of undeclared identifier 'Signal_obj'
Dynamic Signal_obj::__Create(hx::DynamicArray inArgs)
        ^
./src/signal/Signal.cpp:32:6: error: use of undeclared identifier 'Signal_obj'
bool Signal_obj::_hx_isInstanceOf(int inClassId) {
     ^
./src/signal/Signal.cpp:40:6: error: use of undeclared identifier 'Signal_obj'
void Signal_obj::dispatch(){
     ^
./src/signal/Signal.cpp:81:25: error: unknown type name 'Signal_obj'
HX_DEFINE_DYNAMIC_FUNC0(Signal_obj,dispatch,(void))
                        ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
Failed to invoke `haxelib run-dir hxcpp /Users/aidanlee/haxe/haxe_libraries/hxcpp/4.0.0/github/978797989cccff7798b995888a5060388c4d1b76 Build.xml haxe -Dhaxe3=1 -Dhaxe4=1 -Dhaxe_ver=4.000 -Dhxcpp_api_level=400 -Dsignals=1.0.9 '-Dsource-header=Generated by Haxe 4.0.0-rc.2+77068e1' -Dstatic=1 -Dtarget.name=cpp -Dtarget.static=true -Dtarget.sys=true -Dtarget.threaded=true -Dtarget.utf16=true -Dutf16=1 -I/Users/aidanlee/haxe/haxe_libraries/signals/1.0.9/github/4026c0b6adf7e3801a026945a1004487d9c73cad/src/ -Isrc/ -I -I/Users/aidanlee/haxe/versions/4.0.0-rc.2/std/cpp/_std/ -I/Users/aidanlee/haxe/versions/4.0.0-rc.2/std/` because Error: Command failed: haxelib run-dir hxcpp /Users/aidanlee/haxe/haxe_libraries/hxcpp/4.0.0/github/978797989cccff7798b995888a5060388c4d1b76 Build.xml haxe -Dhaxe3=1 -Dhaxe4=1 -Dhaxe_ver=4.000 -Dhxcpp_api_level=400 -Dsignals=1.0.9 '-Dsource-header=Generated by Haxe 4.0.0-rc.2+77068e1' -Dstatic=1 -Dtarget.name=cpp -Dtarget.static=true -Dtarget.sys=true -Dtarget.threaded=true -Dtarget.utf16=true -Dutf16=1 -I/Users/aidanlee/haxe/haxe_libraries/signals/1.0.9/github/4026c0b6adf7e3801a026945a1004487d9c73cad/src/ -Isrc/ -I -I/Users/aidanlee/haxe/versions/4.0.0-rc.2/std/cpp/_std/ -I/Users/aidanlee/haxe/versions/4.0.0-rc.2/std/
Error: Build failed
```
The issue seems to be that the OSX SDK already defines the signal namespace and hxcpp converts haxe packages into cpp namespaces. The above sample will compile fine on OSX when using my fork where I renamed to haxe packages from `signal` to `petesignals`. I've had no issues with the library and hxcpp on Windows and Linux.

Cheers.

**Comment by peteshand** [comment]
A new build has been pushed to github and have been released on haxelib (version 1.1.0) to address the osx cpp package incompatibilities issues. This patch redirects classes in the "signal" package to new classes in the "signals" package. This will allow for backwards compatibility until a future release where the "signal" package will be deprecated.

**Comment by Aidan63** [comment]
Thanks! Just switched my projects back over to this repo and the mac cpp builds are working fine.
